meetings/2021-07/july-13.md

```diff
@@ -0,0 +1,512 @@
+
+Greg: I have a fundamental issue with a non technical argument being something that walks progress of a proposal because there's nothing precluding us from unblocking like that's essentially the argument against objects. 
+
+JHD: Direct object access is also in non-technical one. So that seems to be the situation We're in.
+
+MM:  The argument against direct opposite object access, whether you agree with it or not is a technical argument. the technical argument, is that Shu, You're the one who is the closest that I know of to a representative in the room of the objection. So, please correct me if I'm characterizing the objection wrongly here, but the objection as I understand it, is that if forums are provided, people will try to use it as effectively and isolation mechanism. Call it a security mechanism. If you wish experience and I'll agree with this from the Agoric experience, trying to do, ad hoc isolation on around boundary, with direct object access shows that it is terribly easy to get wrong. It's incredibly easy to make a mistake, such that objects leak, that can be solved either by having well debuggable available libraries like membranes or like your user level calledable boundaries. but the objection is people will underestimate the problem. preserving isolation with ad hoc libraries at the boundary, people will get it wrong. And we will then have security bugs as a result of not having had an enforced reliable isolation
+
+JHD: I'll try and speak slowly maybe not too slowly, had four minutes left shame. So we basically switched from the “only has the foot gun / capability” version to an “Only lacks the foot gun / capability” version. The other option that like, you know, that I would be more interested in is something that defaults to the safer, more restrictive one and allows opting in to the Wider, more powerful, but, you know, easier to screw up version. To me, that would be that would solve all of the use cases presented, my guess is - I know without having that version available and regathering, the evidence that Shu was talking about. I don't know how to confirm that guess. But my guess is that if we named everything, appropriately, whatever that means that we would be able to convey the risk and reduce the foot gun, likelihood that. So that's what I'm interested in seeing. 
```

Comment:
```suggestion
JHD: So we basically switched from the “only has the foot gun / capability” version to an “Only lacks the foot gun / capability” version. The other option that I would be more interested in is something that defaults to the safer, more restrictive one and allows opting in to the wider, more powerful, but easier to screw up version. To me, that would solve all of the use cases presented. I know without having that version available and regathering the evidence that SYG was talking about. I don't know how to confirm that guess. But my guess is that if we named everything appropriately, whatever that means, that we would be able to convey the risk and reduce the footgun likelihood. So that's what I'm interested in seeing.
```